editor/project_manager.cpp

```diff
@@ -427,233 +446,191 @@ void ProjectDialog::_renderer_selected() {
-				} else {
-					// Store default project icon in SVG format.
-					Error err;
-					Ref<FileAccess> fa_icon = FileAccess::open(dir.path_join("icon.svg"), FileAccess::WRITE, &err);
-					fa_icon->store_string(get_default_project_icon());
+		Error err = ProjectSettings::get_singleton()->save_custom(path.path_join("project.godot"), initial_settings, Vector<String>(), false);
+		if (err != OK) {
+			_set_message(TTR("Couldn't create project.godot in project path."), MESSAGE_ERROR);
+			return;
+		}
 
-					if (err != OK) {
-						_set_message(TTR("Couldn't create icon.svg in project path."), MESSAGE_ERROR);
-					}
+		// Store default project icon in SVG format.
+		Ref<FileAccess> fa_icon = FileAccess::open(path.path_join("icon.svg"), FileAccess::WRITE, &err);
+		if (err != OK) {
+			_set_message(TTR("Couldn't create icon.svg in project path."), MESSAGE_ERROR);
+			return;
+		}
+		fa_icon->store_string(get_default_project_icon());
 
-					EditorVCSInterface::create_vcs_metadata_files(EditorVCSInterface::VCSMetadata(vcs_metadata_selection->get_selected()), dir);
-				}
-			} else if (mode == MODE_INSTALL) {
-				if (project_path->get_text().ends_with(".zip")) {
-					dir = install_path->get_text();
-					zip_path = project_path->get_text();
-				}
+		EditorVCSInterface::create_vcs_metadata_files(EditorVCSInterface::VCSMetadata(vcs_metadata_selection->get_selected()), path);
+	} else if ((mode == MODE_IMPORT && path.get_extension() == "zip") || mode == MODE_INSTALL) {
+		if (path.get_extension() == "zip") {
+			zip_path = project_path->get_text();
+			path = install_path->get_text();
+		}
 
-				Ref<FileAccess> io_fa;
-				zlib_filefunc_def io = zipio_create_io(&io_fa);
+		if (create_dir->is_pressed()) {
+			Ref<DirAccess> d = DirAccess::create(DirAccess::ACCESS_FILESYSTEM);
+			if (!d->dir_exists(path) && d->make_dir(path) != OK) {
+				_set_message(TTR("Couldn't create project directory, check permissions."), MESSAGE_ERROR);
+				return;
+			}
+		}
 
-				unzFile pkg = unzOpen2(zip_path.utf8().get_data(), &io);
-				if (!pkg) {
-					dialog_error->set_text(TTR("Error opening package file, not in ZIP format."));
-					dialog_error->popup_centered();
-					return;
-				}
+		Ref<FileAccess> io_fa;
+		zlib_filefunc_def io = zipio_create_io(&io_fa);
 
-				// Find the zip_root
-				String zip_root;
-				int ret = unzGoToFirstFile(pkg);
-				while (ret == UNZ_OK) {
-					unz_file_info info;
-					char fname[16384];
-					unzGetCurrentFileInfo(pkg, &info, fname, 16384, nullptr, 0, nullptr, 0);
-
-					String name = String::utf8(fname);
-					if (name.ends_with("project.godot")) {
-						zip_root = name.substr(0, name.rfind("project.godot"));
-						break;
-					}
+		unzFile pkg = unzOpen2(zip_path.utf8().get_data(), &io);
+		if (!pkg) {
+			_set_message(TTR("Error opening package file, not in ZIP format."), MESSAGE_ERROR);
+			return;
+		}
 
-					ret = unzGoToNextFile(pkg);
-				}
+		// Find the zip_root
```

Comment:
```suggestion
		// Find the zip_root.
```
Per the [comment style](https://docs.godotengine.org/en/latest/contributing/development/code_style_guidelines.html#comment-style-guide).